lib/containers/devices_container.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:lite/models/model.dart';
+import 'package:lite/redux/actions.dart';
+import 'package:lite/models/device_model.dart';
+import 'package:lite/screens/device_list_screen.dart';
+import 'package:redux/redux.dart';
+
+class DevicesContainer extends StatelessWidget {
+
+  @override
+  Widget build(BuildContext context) {
+    return StoreConnector<AppState, _ViewModel>(
+      // convert the store into a viewModel
+      // the resulting viewModel will be passed into the builder function
+        converter: _ViewModel.fromStore,
+        builder: (context, vm) =>
+            DeviceListScreen(vm.deviceList, vm.config, vm.lightsStateCache, vm.onToggle)
+    );
+  }
+}
+
+class _ViewModel {
+  List<DeviceModel> deviceList;
+  Config config;
+  Map<String, bool> lightsStateCache;
+  Function(String) onToggle;
+
+  _ViewModel(this.deviceList, this.config, this.lightsStateCache, this.onToggle);
+
+  static _ViewModel fromStore(Store<AppState> store) {
```

Comment:
Consider if there's a better place to dispatch to trigger middleware async calls based on state - reducers have access to state, but not to `store.dispatch`.